src/main/java/com/example/audit/core/entity/Admin.java

```diff
@@ -0,0 +1,48 @@
+package com.example.audit.core.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import javax.persistence.Entity;
+import javax.persistence.Table;
+import java.util.Collection;
+
+@Data
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+@Table(name = "admin")
+@EqualsAndHashCode(callSuper = true)
+public class Admin extends AbstractEntity  implements UserDetails {
```

Comment:
it is not a good idea to have a separate entity for the admin user
you already have users in the system with the admin role, the authentication should work across all the services

Reply:
for now I am ok with that but it should be reworked later with single authentication service